### PR TITLE
External orchestration support

### PR DESCRIPTION
**Pull request by 3Hren**
- [x] Drop external balancing stuff
 - [x] Pluggable external orchestrator
 - [x] Refactoring
  - [x] Use `boost::algorithms` instead of `min_element_if`
  - [x] Prettify
  - [x] Logging.
 - [x] No more than once control channel per app.

### CODE REVIEW

**Comment by 3Hren** [comment]
Okay, now everyone (@antmat, @noxiouz, @kobolog), take a look, please.

**Comment by 3Hren** [comment]
Still waiting for a review.

**Comment by noxiouz** [comment]
Sorry, but I wanna try first :)

File: node/src/node/overseer.cpp [hunk 1]
```diff
@@ -459,5 +444,124 @@ overseer_t::on_slave_death(const std::error_code& ec, std::string uuid) {
...
+    using boost::adaptors::filtered;
+
+    const auto load = queue->size();
+    const auto profile = this->profile();
+
+    const auto pool_target = this->pool_target.load();
+
+    // Bound current pool target between [1; limit].
+    const auto target = detail::bound(
+        1UL,
```
Comment (noxiouz) [inline node/src/node/overseer.cpp]: Seems it makes impossible to turn off all workers via `control`.

Comment (3Hren) [inline node/src/node/overseer.cpp]: Yep.

Comment (kobolog) [inline node/src/node/overseer.cpp]: Should it be possible?

Comment (3Hren) [inline node/src/node/overseer.cpp]: What should I do if there are 0 workers allowed and a new request comes? Drop? Put in the queue? Why did this request come at all?

**Pull request merged**